feat: Updated and added examples (logs and metrics)

Jayclifford345 opened the pull request:
# Description

Small PR to update some of the examples we have within the docs folder.
1. Fixed logs example (collector config errored out due to a bad configuration parameter. The debug component only supports: `(detailed|normal|basic)`
1. Added example of adding custom attributes to an event in `docs/examples/logs/example.py`
1. Added example for using `PeriodicExportingMetricReader` to explicitly collect metrics synchronously.  This is to provide an example of the workaround for #4549

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

- Run linting
- Run examples

# Does This PR Require a Contrib Repo Change?

<!--
Answer the following question based on these examples of changes that would require a Contrib Repo Change:
- [The OTel specification](https://github.com/open-telemetry/opentelemetry-specification) has changed which prompted this PR to update the method interfaces of `opentelemetry-api/` or `opentelemetry-sdk/`
- The method interfaces of `test/util` have changed
- Scripts in `scripts/` that were copied over to the Contrib repo have changed
- Configuration files that were copied over to the Contrib repo have changed (when consistency between repositories is applicable) such as in
    - `pyproject.toml`
    - `isort.cfg`
    - `.flake8`
- When a new `.github/CODEOWNER` is added
- Major changes to project information, such as in:
    - `README.md`
    - `CONTRIBUTING.md`
-->

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [x] Documentation has been updated

emdneto commented on docs/examples/metrics/reader/synchronous_guage_read.py:
```diff
@@ -0,0 +1,88 @@
+
+guage = meter.create_observable_gauge(
+    name="synchronous_gauge_temperature",
+    description="Gauage value captured synchronously",
```
```suggestion
    description="Gauge value captured synchronously",
```

emdneto commented on docs/examples/metrics/reader/synchronous_guage_read.py:
```diff
@@ -0,0 +1,88 @@
+# Invoking `collect` will read all measurements assigned to the reader
+guage2 = meter.create_observable_gauge(
+    name="synchronous_gauge_humidity",
+    description="Gauage value captured synchronously",
```
```suggestion
    description="Gauge value captured synchronously",
```

Jayclifford345 commented:
Hi @emdneto,
Apologies for the spelling errors there, I have updated them and run
```
tox -e spellcheck
```
So should have caught all now. The CHANGELOG test failed, so I added it to the PR to the unreleased section, but thought it was such a minor alteration that it wouldn't be needed ,so happy to remove if you like

emdneto commented on docs/examples/metrics/reader/README.rst:
```diff
@@ -6,6 +6,7 @@ These examples show how to customize the metrics that are output by the SDK usin
 * preferred_aggregation.py: Shows how to configure the preferred aggregation for metric instrument types.
 * preferred_temporality.py: Shows how to configure the preferred temporality for metric instrument types.
 * preferred_exemplarfilter.py: Shows how to configure the exemplar filter.
+* synchronous_gauge_reader.py: Shows how to use `PeriodicExportingMetricReader` in a synchronous manner to explicitly control the collection of metrics.
```
```suggestion
* synchronous_gauge_read.py: Shows how to use `PeriodicExportingMetricReader` in a synchronous manner to explicitly control the collection of metrics.
```

emdneto reviewed (commented):
Thanks. one more change. I think we don't need changelog, but you've already added so it's up to you. If you want, I can add the label to skip it

Jayclifford345 commented:
Happy to work on more examples / docs in the future so if you have any in the backlog after this just let me know :)
